MSAL/src/MSALErrorConverter.m

```diff
@@ -44,29 +44,35 @@ + (void)initialize
+                                   @(MSIDErrorUnsupportedFunctionality): @(MSALErrorInternal),
+                                   // Cache
                                    @(MSIDErrorCacheMultipleUsers) : @(MSALErrorInternal),
-                                   @(MSIDErrorTokenCacheItemFailure) : @(MSALErrorTokenCacheItemFailure),
-                                   @(MSIDErrorWrapperCacheFailure) : @(MSALErrorWrapperCacheFailure),
                                    @(MSIDErrorCacheBadFormat) : @(MSALErrorWrapperCacheFailure),
-                                   @(MSIDErrorCacheVersionMismatch) : @(MSALErrorInternal),
-                                   @(MSIDErrorServerInvalidResponse) : @(MSALErrorInvalidResponse),
-                                   @(MSIDErrorDeveloperAuthorityValidation) : @(MSALErrorFailedAuthorityValidation),
-                                   @(MSIDErrorServerRefreshTokenRejected) : @(MSALErrorAuthorizationFailed),
-                                   @(MSIDErrorServerOauth) : @(MSALErrorAuthorizationFailed),
-                                   @(MSIDErrorUnsupportedFunctionality): @(MSALErrorInternal)
+                                   // Authority Validation
+                                   @(MSIDErrorAuthorityValidation) : @(MSALErrorFailedAuthorityValidation),
+                                   // Interactive flow
+                                   @(MSIDErrorAuthorizationFailed) : @(MSALErrorAuthorizationFailed),
+                                   @(MSIDErrorUserCancel) : @(MSALErrorUserCanceled),
+                                   @(MSIDErrorSessionCanceledProgramatically) : @(MSALErrorSessionCanceled),
+                                   @(MSIDErrorInteractiveSessionStartFailure) : @(MSALErrorInternal),
+                                   @(MSIDErrorInteractiveSessionAlreadyRunning) : @(MSALErrorInteractiveSessionAlreadyRunning),
+                                   @(MSIDErrorNoMainViewController) : @(MSALErrorNoViewController),
                                    },
                            MSIDOAuthErrorDomain:@{
-                                   @(MSIDErrorInvalidRequest) :@(MSALErrorInvalidRequest),
-                                   @(MSIDErrorInvalidClient) : @(MSALErrorInvalidClient),
-                                   @(MSIDErrorInvalidGrant) : @(MSALErrorInvalidParameter),
-                                   @(MSIDErrorInvalidParameter) : @(MSALErrorInvalidParameter),
-                                   @(MSIDErrorServerRefreshTokenRejected) : @(MSALErrorAuthorizationFailed),
+                                   @(MSIDErrorInteractionRequired) : @(MSALErrorInteractionRequired),
                                    @(MSIDErrorServerOauth) : @(MSALErrorAuthorizationFailed),
+                                   @(MSIDErrorServerInvalidResponse) : @(MSALErrorInvalidResponse),
+                                   @(MSIDErrorServerRefreshTokenRejected) : @(MSALErrorAuthorizationFailed),
```

Comment:
Is there a better corresponding error code than MSALErrorAuthorizationFailed (which sounds more like interactive error)?

Reply:
added corresponding error for msal. also filed an issue to improve refactoring #285